### PR TITLE
Organize folder structure

### PR DESCRIPTION
**Pull request by bsudekum**
image

/cc @1ec5 @frederoni

### CODE REVIEW

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -563,6 +554,31 @@
...
 		};
+		C51923B41EA55C5E002AF9E1 /* Extensions */ = {
+			isa = PBXGroup;
+			children = (
+				351BEC081E5BCC72006FE110 /* Bundle.swift */,
+				351BEC0A1E5BCC72006FE110 /* Directions.swift */,
+				351BEC041E5BCC6C006FE110 /* ManeuverDirection.swift */,
+				351BEBEB1E5BCC63006FE110 /* String.swift */,
+				351BEBF01E5BCC63006FE110 /* UIView.swift */,
+			);
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: MGLMapView+MGLNavigationAdditions.{h,m} is also an extension (or category, as it’s called in Objective-C).

Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: MGLMapView.swift also consists of an extension.

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -563,6 +554,31 @@
...
+			isa = PBXGroup;
+			children = (
+				351BEC091E5BCC72006FE110 /* DashedLineView.swift */,
+				351BEC031E5BCC6C006FE110 /* LaneArrowView.swift */,
+				351BEBEC1E5BCC63006FE110 /* StyleButton.swift */,
+				351BEBED1E5BCC63006FE110 /* StyleKitArrows.swift */,
+				351BEBEE1E5BCC63006FE110 /* StyleLabel.swift */,
+				351BEBEF1E5BCC63006FE110 /* TurnArrowView.swift */,
+			);
+			name = "UI Style";
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: How about “Views”? This group contains views (and some classes that implement custom drawing of views) that aren’t part of the route table.

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -563,6 +554,31 @@
...
+			sourceTree = "<group>";
+		};
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: The various RouteTable*.swift files could be a Route Table group as well.

File: MapboxNavigation.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -430,34 +430,25 @@
 		351BEBD81E5BCC28006FE110 /* MapboxNavigation */ = {
 			isa = PBXGroup;
 			children = (
+				C51923B41EA55C5E002AF9E1 /* Extensions */,
+				351BEC201E5BD4DC006FE110 /* Supporting files */,
```
Comment (1ec5) [inline MapboxNavigation.xcodeproj/project.pbxproj]: Nit: Title Case

Comment (bsudekum) [inline MapboxNavigation.xcodeproj/project.pbxproj]: heh, pretty certain folder is created by xcode but 🤷‍♀️

**Pull request merged**